### PR TITLE
[LP-2623] Integrate webinars on learn page

### PR DESCRIPTION
**Pull request by muak50**

### CODE REVIEW

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1528/workflows/workflows_id/jobs/1528/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
No lines with coverage information in this diff.
-------------
```

**Comment by muak50** [comment]
> **pylint:** ❌
> **pycodestyle:** ✔️
> **iSort:** ❌
> 
> **Please fix following pylint issues:**
> 
> ```
> ************* Module edx-platform.lms.djangoapps.courseware.views.views
> lms/djangoapps/courseware/views/views.py:1711:0: C0301: Line too long (121/120) (line-too-long)
> lms/djangoapps/courseware/views/views.py:1727:0: C0301: Line too long (123/120) (line-too-long)
> lms/djangoapps/courseware/views/views.py:1728:0: C0301: Line too long (161/120) (line-too-long)
> lms/djangoapps/courseware/views/views.py:1730:0: C0301: Line too long (165/120) (line-too-long)
> lms/djangoapps/courseware/views/views.py:489:8: W0611: Unused CourseReviewsModuleFragmentView imported from openedx.features.course_experience.views.course_reviews (unused-import)
> lms/djangoapps/courseware/views/views.py:564:4: W0221: Parameters differ from overridden 'get' method (arguments-differ)
> lms/djangoapps/courseware/views/views.py:579:4: W0221: Parameters differ from overridden 'render_to_fragment' method (arguments-differ)
> lms/djangoapps/courseware/views/views.py:585:4: W0221: Parameters differ from overridden 'render_standalone_response' method (arguments-differ)
> lms/djangoapps/courseware/views/views.py:605:4: W0221: Parameters differ from overridden 'get' method (arguments-differ)
> lms/djangoapps/courseware/views/views.py:702:11: R1701: Consider merging these isinstance calls to isinstance(exception, (Http404, Redirect)) (consider-merging-isinstance)
> lms/djangoapps/courseware/views/views.py:703:12: E0704: The raise statement is not inside an except clause (misplaced-bare-raise)
> lms/djangoapps/courseware/views/views.py:705:12: E0704: The raise statement is not inside an except clause (misplaced-bare-raise)
> lms/djangoapps/courseware/views/views.py:768:4: W0221: Parameters differ from overridden 'render_to_fragment' method (arguments-differ)
> lms/djangoapps/courseware/views/views.py:775:4: W0221: Parameters differ from overridden 'render_standalone_response' method (arguments-differ)
> lms/djangoapps/courseware/views/views.py:930:8: W0612: Unused variable 'registration_price' (unused-variable)
> lms/djangoapps/courseware/views/views.py:1031:4: W0611: Unused COURSE_DATES_NAME imported from lms.urls (unused-import)
> lms/djangoapps/courseware/views/views.py:1198:0: C0116: Missing function or method docstring (missing-function-docstring)
> lms/djangoapps/courseware/views/views.py:1219:0: C0116: Missing function or method docstring (missing-function-docstring)
> lms/djangoapps/courseware/views/views.py:1505:11: R5101: Instead of HttpResponse(json.dumps(data)) use JsonResponse(data) (http-response-with-json-dumps)
> lms/djangoapps/courseware/views/views.py:1505:11: R5102: Instead of HttpResponse(content_type='application/json') use JsonResponse() (http-response-with-content-type-json)
> lms/djangoapps/courseware/views/views.py:1641:4: W0611: Unused COURSE_DATES_NAME imported from lms.urls (unused-import)
> lms/djangoapps/courseware/views/views.py:1803:12: R1714: Consider merging these comparisons with "in" to 'zendesk_submitted in (200, 201)' (consider-using-in)
> lms/djangoapps/courseware/views/views.py:1821:17: E7610: i18n function _() must be called with a literal string (translation-of-non-string)
> lms/djangoapps/courseware/views/views.py:1854:25: E7610: i18n function _() must be called with a literal string (translation-of-non-string)
> lms/djangoapps/courseware/views/views.py:1864:25: E7610: i18n function _() must be called with a literal string (translation-of-non-string)
> lms/djangoapps/courseware/views/views.py:1877:25: E7610: i18n function _() must be called with a literal string (translation-of-non-string)
> lms/djangoapps/courseware/views/views.py:1890:25: E7610: i18n function _() must be called with a literal string (translation-of-non-string)
> lms/djangoapps/courseware/views/views.py:102:0: W0611: Unused get_enrollment imported from openedx.core.djangoapps.enrollments.api (unused-import)
> 
> -----------------------------------
> Your code has been rated at 9.19/10
> ```
> 
> **Please fix following iSort issues:**
> 
> ```
>  ERROR: /lms/djangoapps/courseware/views/views.py Imports are incorrectly sorted. 
> ```

Pylint and isort issues are related to core code.

File: lms/djangoapps/courseware/views/views.py [hunk 1]
```diff
@@ -267,12 +268,16 @@ def courses(request):
     # Add marketable programs to the context.
     programs_list = get_programs_with_type(request.site, include_hidden=False)
 
+    # Get upcoming webinars to display them on `learn` page
+    upcoming_webinars = Webinar.objects.filter(status=Webinar.UPCOMING).order_by('start_time')
```
Comment (javeriaejaz-arbisoft) [inline lms/djangoapps/courseware/views/views.py:272]: Shouldn't it be descending?

**Review by aliassad — APPROVED** [review]
LGTM :+1

File: lms/djangoapps/courseware/views/views.py [hunk 2]
```diff
@@ -267,12 +268,16 @@ def courses(request):
...
+
```
Comment (asadali145) [inline lms/djangoapps/courseware/views/views.py:273]: No need for this variable can be moved directly at line 280 to avoid core changes at 2 places.

**Review by javeriaejaz-arbisoft — APPROVED** [review]

File: lms/djangoapps/courseware/views/views.py (hunk 1 above)
Comment (muak50) [reply lms/djangoapps/courseware/views/views.py:272]: No, we need to display them in [ascending order](https://philanthropyu.atlassian.net/browse/LP-2623)

File: lms/djangoapps/courseware/views/views.py (hunk 2 above)
Comment (muak50) [reply lms/djangoapps/courseware/views/views.py:273]: Sure, I'll do that while creating PR for master.

Comment (asadali145) [reply lms/djangoapps/courseware/views/views.py:273]: Create a PR for develop as well to merge that commit in develop as well.

**Pull request merged**